src/create-model.jl

```diff
@@ -50,20 +52,56 @@ function construct_dataframes(graph, representative_periods, constraints_partiti
         dataframes[key] = df
     end
 
+    # Dataframe to store the storage level between (inter) representative period variable (a.k.a. long-term or seasonal storage)
+    # create base_periods_block_per_asset
+    base_periods_block_per_asset = Dict(
+        a => begin
+            n = base_periods.num_base_periods
+            m = graph[a].moving_window_long_storage
+            [(1+m*(j-1)):min(n, m * j) for j = 1:ceil(Int, n / m)]
+        end for
+        a in A if graph[a].type == "storage" && coalesce(graph[a].storage_type == "long", false)
+    )
+
+    if !isempty(base_periods_block_per_asset)
+        dataframes[:storage_level_inter_rp] = DataFrame(
+            (
+                (
+                    (asset = a, base_period_block = base_period_block) for
+                    base_period_block in base_period_blocks
+                ) for (a, base_period_blocks) in base_periods_block_per_asset
+            ) |> Iterators.flatten,
+        )
+        dataframes[:storage_level_inter_rp].index = 1:size(dataframes[:storage_level_inter_rp], 1)
+    else
+        # No data, but ensure schema is correct
+        dataframes[:storage_level_inter_rp] =
+            DataFrame(; asset = String[], base_period_block = Int[], index = Int[])
+    end
+
     return dataframes
 end
 
 """
-    add_expression_terms!(df_cons, df_flows, workspace, representative_periods; add_efficiency)
+    add_expression_terms_intra_rp_constraints!(df_cons,
+                                               df_flows,
+                                               workspace,
+                                               representative_periods,
+                                               graph;
+                                               use_highest_resolution = true,
+                                               multiply_by_duration = true,
+                                               )
+
+Computes the incoming and outgoing expressions per row of df_cons for the constraints
+that are within (intra) the representative period.
 
-Computes the incoming and outgoing expressions per row of df_cons. This function
-is only used internally in the model.
+This function is only used internally in the model.
 
 This strategy is based on the replies in this discourse thread:
 
   - https://discourse.julialang.org/t/help-improving-the-speed-of-a-dataframes-operation/107615/23
 """
-function add_expression_terms!(
+function add_expression_terms_intra_rp_constraints!(
```

Comment:
It's better like that because with this implementation, we have two types of constraints:

1. The ones within (intra) the representative periods (all balances and limits, not only for storage) that have the flexible temporal resolution with the `partitions`.
2. The ones between (inter) representative periods (so far, only the storage balance) that are related to the called `base periods`

So, looking into the future developments in Tulipa, it is better to use this name.

We can extrapolate the idea of the partitions to implement it at the base period level. It will make the code more uniform in the temporal handling. I will add an issue for that but with a priority `Eventually` (since it is not urgent)

I hope I have made my case to keep the name 😄

Reply:
Works for me